sympy/sets/fancysets.py

```diff
@@ -443,3 +445,309 @@ def _sup(self):
+        rectangular form, or the union of intervals of `theta`
+        when self is in polar form.
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, ComplexPlane, Union
+        >>> a = Interval(2, 3)
+        >>> b = Interval(4, 5)
+        >>> c = Interval(1, 7)
+        >>> C1 = ComplexPlane(a*b)
+        >>> C1.b_interval
+        [4, 5]
+        >>> C2 = ComplexPlane(Union(a*b, b*c))
+        >>> C2.b_interval
+        [1, 7]
+
+        """
+        b_interval = []
+        for element in self.psets:
+            b_interval.append(element.args[1])
+
+        b_interval = Union(*b_interval)
+        return b_interval
+
+    @property
+    def polar(self):
+        """
+        Returns True if self is in polar form.
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, ComplexPlane, Union, S
+        >>> a = Interval(2, 3)
+        >>> b = Interval(4, 5)
+        >>> theta = Interval(0, 2*S.Pi)
+        >>> C1 = ComplexPlane(a*b)
+        >>> C1.polar
+        False
+        >>> C2 = ComplexPlane(a*theta, polar=True)
+        >>> C2.polar
+        True
+        """
+        return self.args[0].args[1].is_Mul
+
+    def _contains(self, other):
+        from sympy.functions import arg, Abs
+
+        # self in rectangular form
+        if self.polar is False:
+            re, im = other.as_real_imag()
+            for element in self.psets:
+                if And(element.args[0]._contains(re),
+                        element.args[1]._contains(im)):
+                    return True
+            return False
+
+        # self in polar form
+        elif self.polar:
+            if S(other).is_zero:
+                r, theta = S(0), S(0)
+            else:
+                r, theta = Abs(other), arg(other)
+            for element in self.psets:
+                if And(element.args[0]._contains(r),
+                        element.args[1]._contains(theta)):
+                    return True
+                return False
+
+    def _intersect(self, other):
+
+        if other.is_ComplexPlane:
+            # self in rectangular form
+            if (not self.polar) and (not other.polar):
+                return ComplexPlane(Intersection(self.sets, other.sets))
+
+            # self in polar form
+            elif self.polar and other.polar:
+                r1, theta1 = self.a_interval, self.b_interval
+                r2, theta2 = other.a_interval, other.b_interval
+                new_r_interval = Intersection(r1, r2)
+                new_theta_interval = Intersection(theta1, theta2)
+                # 0 and 2*Pi means the same
+                if (2*S.Pi in theta1 and S(0) in theta2) or (2*S.Pi in theta2 and S(0) in theta1):
+                    new_theta_interval = Union(new_theta_interval, FiniteSet(0))
+                return ComplexPlane(new_r_interval*new_theta_interval,
+                                    polar=True)
+
+        if other.is_Interval:
```

Comment:
So the general idea is to take real values in the complex set and take their intersection with the real set provided. In that case you real set need not be an Interval, it can be any general real set.